core/arch/arm/plat-hikey/main.c

```diff
@@ -205,4 +206,47 @@ static TEE_Result peripherals_init(void)
 	return TEE_SUCCESS;
 }
 
+/*
+ * First time, add low 16 bits of tsc as entropy.
+ * Subsequently, accumulate 2 low bits each time
+ * and add entropy when we reach 8 bits.
+ */
+
+void plat_prng_add_jitter_entropy(void)
+{
+	uint64_t tsc = 0;
+	int bytes = 0;
+	static uint8_t first, bits;
+	static uint16_t acc;
+
+#if defined(CFG_ARM64_core)
+	__asm__("mrs %0, cntvct_el0" : "=r" (tsc));
+#else
+	{
+		uint32_t tsc32;
+
+		__asm__("mrc p15, 0, %0, c9, c13, 0" : "=r"(tsc32));
+		tsc = tsc32;
+	}
+#endif
```

Comment:
Avoid inline assembly in `.c` files. See `core/arch/arm/include/arm{,32,64}.h`.
As I said above, why would this be better than using `tee_time_get_sys_time()` as a source of entropy? `tee_time_get_sys_time()` is portable, and is typically based on CNTPCT/CNTPCT_EL0.